### PR TITLE
Issue #9311 - populate search bar on select

### PR DESCRIPTION
**Pull request by ale-cristofori**
## Description
When performing a search from the map page search bar, results appear and when selecting the desired one the selected text should populate the search bar.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
#9311 

**What is the current behavior?**
The search text box is not populated with the selected item text but with the text typed by the user.
#9311 

**What is the new behavior?**
The search text box should be populated with the selected item text.
https://github.com/geosolutions-it/MapStore2/assets/14953970/9e082e0c-ab2e-421e-8d5b-e4dd6ddc4f94

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/actions/search.js [hunk 1]
```diff
@@ -216,11 +216,12 @@ export function textSearch(searchText, {services = null} = {}, maxResults = 15)
  * @param {object} mapConfig the current map configuration (with size, projection...)
  * @param {object} resultsStyle style to apply to results geometries
  */
-export function selectSearchItem(item, mapConfig, resultsStyle) {
+export function selectSearchItem(item, mapConfig, service, resultsStyle) {
```
Comment (MV88) [inline web/client/actions/search.js:219]: missing js doc

**Review by MV88 — CHANGES_REQUESTED** [review]
LGTM just one little thing

File: web/client/actions/search.js (hunk 1 above)
Comment (ale-cristofori) [reply web/client/actions/search.js:220]: included JSDOC for new parameter

**Review by ale-cristofori — COMMENTED** [review]
re-reviewed and modified as requested

**Review by MV88 — APPROVED** [review]

**Comment by tdipisa** [comment]
@ElenaGallo please test this in DEV.

**Pull request merged**